repose-aggregator/components/filters/api-validator-filter/src/main/java/org/openrepose/filters/apivalidator/ApiValidatorHandler.java

```diff
@@ -153,9 +146,16 @@ public void doFilter(HttpServletRequest request, HttpServletResponse response, F
                 }
 
                 if (!isValid) {
-                    if (mbcsInvalidRequests != null) {
+                    if (Optional.ofNullable(metricsService).isPresent()) {
```

Comment:
You know what? I take that back. `Optional`s are totally the best. Turns out, Spring supports Java 8's `Optional` when it auto-wires. So I propose a solution very similar to yours, distilled into the following example:

```
@Named
class MyFilter @Inject()(Optional[MetricsService] metricsService) extends Filter {

  private val metricRegistry: Optional[MetricsRegistry] = metricsService.map(MetricsService::getRegistry)

  ...

  override def doFilter(request: ServletRequest, response: ServletResponse, chain: FilterChain): Unit = {
    metricsRegistry.ifPresent(mr -> mr.meter("myMeter").mark())
  }

  ...

}
```

Note that, yes, I am using a Java `Optional` in Scala.